bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -0,0 +1,719 @@
+	private final AccountFileInstance selfAccountInstance;
+
+	// type of accounts to be created for the Application using this helper
+	private final Account.Type accountsType;
+
+	/**
+	 * Constructs a AppDB parser helper for the given DB file.
+	 *
+	 * This is a constructor for Apps that that do not have any app specific
+	 * account information for device owner and will use a 'Device' account in
+	 * lieu.
+	 *
+	 * It creates a DeviceAccount instance to use as a self account.
+	 *
+	 * @param caseDb       Sleuthkit case db
+	 * @param moduleName   name module using the helper
+	 * @param srcFile      source file being processed by the module
+	 * @param accountsType account types created by this module
+	 *
+	 * @throws TskCoreException
+	 */
+	public CommunicationArtifactsHelper(SleuthkitCase caseDb, String moduleName, AbstractFile srcFile, Account.Type accountsType) throws TskCoreException {
+
+		super(caseDb, moduleName, srcFile);
+
+		this.accountsType = accountsType;
+		this.selfAccountInstance = getSleuthkitCase().getCommunicationsManager().createAccountFileInstance(Account.Type.DEVICE, ((DataSource) getAbstractFile().getDataSource()).getDeviceId(), moduleName, getAbstractFile());
+	}
+
+	/**
+	 * Constructs a AppDB parser helper for the given DB file.
+	 *
+	 * This constructor is for Apps that do have app specific account
+	 * information for the device owner to create a 'self' account.
+	 *
+	 * It creates a an account instance with specified type & id and uses it as
+	 * a self account.
+	 *
+	 * @param caseDb             Sleuthkit case db
+	 * @param moduleName         name module using the helper
+	 * @param srcFile            source file being processed by the module
+	 * @param accountsType       account types created by this module
+	 * @param selfAccountType    self account type to be created for this module
+	 * @param selfAccountAddress account unique id for the self account
+	 *
+	 * @throws TskCoreException
+	 */
+	public CommunicationArtifactsHelper(SleuthkitCase caseDb, String moduleName, AbstractFile srcFile, Account.Type accountsType, Account.Type selfAccountType, Account.Address selfAccountAddress) throws TskCoreException {
+
+		super(caseDb, moduleName, srcFile);
+
+		this.accountsType = accountsType;
+		this.selfAccountInstance = getSleuthkitCase().getCommunicationsManager().createAccountFileInstance(selfAccountType, selfAccountAddress.getUniqueID(), moduleName, getAbstractFile());
+	}
+
+	/**
+	 * Creates and adds a TSK_CONTACT artifact to the case, with specified
+	 * attributes. Also creates an account instance of specified type for the
+	 * contact with the specified ID.
+	 *
+	 * @param contactAccountUniqueID unique id for contact account
+	 * @param contactName            Name of contact, required
+	 * @param phoneNumber            primary phone number for contact, may be
+	 *                               empty or null
+	 * @param homePhoneNumber        home phone number, may be empty or null
```

Comment:
This is public documentation and should be polished. For example, we should either use lower case or upper case for parameter documentation. Please be consistent. When I'm reading documentation, lack of consistency suggests lack of attention to detail. I prefer initial upper case - perhaps this should become a project coding guideline.